### PR TITLE
Show unapplied function references in call hierarchy

### PR DESCRIPTION
**Pull request by ahoppen**
With https://github.com/apple/swift/pull/72930 we can use the `containedBy` instead of `calledBy` relation for the call hierarchy, which allows us to show unapplied function references in the call hierarchy as well.

rdar://123769825

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Tests/SourceKitLSPTests/CallHierarchyTests.swift [hunk 1]
```diff
@@ -606,4 +608,88 @@ final class CallHierarchyTests: XCTestCase {
...
+              "usr": .string("s:4test3fooyyF"),
+              "uri": .string(project.fileURI.stringValue),
+            ])
+          ),
+          fromRanges: [Range(project.positions["3️⃣"])]
+        )
+      ]
+    )
+  }
 }
```
Comment (bnbarham) [inline Tests/SourceKitLSPTests/CallHierarchyTests.swift:695]: Could you add a test case for a stored property initialized with a closure?

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**